ddtrace/contrib/boto/patch.py

```diff
@@ -47,31 +63,36 @@ def patched_query_request(original_func, instance, args, kwargs):
     if not pin or not pin.enabled():
         return original_func(*args, **kwargs)
 
-    endpoint_name = getattr(instance, "host").split('.')[0]
+    endpoint_name = getattr(instance, "host").split(".")[0]
 
-    with pin.tracer.trace('{}.command'.format(endpoint_name), service="{}.{}".format(pin.service, endpoint_name),
-                          span_type=SPAN_TYPE) as span:
+    with pin.tracer.trace(
+        "{}.command".format(endpoint_name),
+        service="{}.{}".format(pin.service, endpoint_name),
+        span_type=SPAN_TYPE,
+    ) as span:
 
         operation_name = None
         if args:
             operation_name = args[0]
-            span.resource = '%s.%s' % (endpoint_name, operation_name.lower())
+            span.resource = "%s.%s" % (endpoint_name, operation_name.lower())
         else:
             span.resource = endpoint_name
 
         # Adding the args in AWS_QUERY_TRACED_ARGS if exist to the span
         if not aws.is_blacklist(endpoint_name):
-            for arg in aws.unpacking_args(args, AWS_QUERY_ARGS_NAME, AWS_QUERY_TRACED_ARGS):
+            for arg in aws.unpacking_args(
+                args, AWS_QUERY_ARGS_NAME, AWS_QUERY_TRACED_ARGS
+            ):
                 span.set_tag(arg[0], arg[1])
 
         # Obtaining region name
-        region = getattr(instance, "region")
+        region = getattr(instance, "region", None)
         region_name = get_region_name(region)
 
         meta = {
-            'aws.agent': 'boto',
-            'aws.operation': operation_name,
-            'aws.region': region_name,
+            "aws.agent": "boto",
+            "aws.operation": operation_name,
+            "aws.region": region_name,
```

Comment:
with the `None` default, `get_region_name(region)` returns `None`. Maybe we should add `aws.region` tag only if it's not none?

Reply:
I'd prefer this behaviour, but I opted to stay consistent with the functionality of line 143:
https://github.com/DataDog/dd-trace-py/blob/56cb63f2bc16a70bfe08174f751d4453afcc3566/ddtrace/contrib/boto/patch.py#L143

Reply:
This will result in the region tag showing up as `'None'`, which isn't ideal. Let's not add it if it's none.

Reply:
exactly, let's not add it if it's `None`